spec/models/waste_exemptions_engine/transient_registration_workflow_state/business_type_form_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteExemptionsEngine
+  RSpec.describe TransientRegistration, type: :model do
+    describe "#workflow_state" do
+      previous_state = :applicant_email_form
+      current_state = :business_type_form
+      subject(:transient_registration) { create(:transient_registration, workflow_state: current_state) }
+
+      context "when a TransientRegistration's state is #{current_state}" do
+        context "when neither transient_registration.partnership? nor " \
+        "transient_registration.skip_registration_number? are true" do
+          next_state = :registration_number_form
+
+          %w[limitedCompany limitedLiabilityPartnership].each do |business_type|
+            before(:each) { transient_registration.business_type = business_type }
+
+            context "and the business type is #{business_type}" do
+              it "can only transition to either #{previous_state} or #{next_state}" do
+                permitted_states = transient_registration.aasm.states(permitted: true).map(&:name)
```

Comment:
Already had a brief discussion about this. It's one line and readable as it is, but there is a certain amount of complexity in it that is duplicated throughout these tests.

So I think there would be some value in moving this code to a test helper, and referring to that in the tests instead.

Reply:
I think that might be useful as well, since it's used so much.